### PR TITLE
chore(merge-tree sequence): remove more public exports

### PR DESCRIPTION
**Pull request by connorskees**
Removes from the public exports or marks `@internal` a number of types and functions from merge-tree and sequence.

I'm hoping for this to be the last change to the merge-tree public API for this release cycle.

<details>

<summary>The stable, non-internal API surface of merge-tree</summary>

 - `Client`
 - `createDetachedLocalReferencePosition`
 - `LocalReferenceCollection`
 - `LocalReferencePosition`
 - `SlidingPreference`
 - `IMergeTreeAttributionOptions`
 - `IMergeTreeOptions`
 - `IMergeTreeClientSequenceArgs`
 - `IMergeTreeDeltaCallbackArgs`
 - `IMergeTreeDeltaOpArgs`
 - `IMergeTreeMaintenanceCallbackArgs`
 - `IMergeTreeSegmentDelta`
 - `MergeTreeDeltaOperationType`
 - `MergeTreeDeltaOperationTypes`
 - `MergeTreeMaintenanceType`
 - `BaseSegment`
 - `CollaborationWindow`
 - `debugMarkerToString`
 - `IJSONMarkerSegment`
 - `IMergeNodeCommon`
 - `IRemovalInfo`
 - `ISegment`
 - `ISegmentAction`
 - `Marker`
 - `MergeNode`
 - `reservedMarkerIdKey`
 - `Trackable`
 - `TrackingGroup`
 - `ITrackingGroup`
 - `TrackingGroupCollection`
 - `ICombiningOp`
 - `IJSONSegment`
 - `IMarkerDef`
 - `IMergeTreeAnnotateMsg`
 - `IMergeTreeDelta`
 - `IMergeTreeDeltaOp`
 - `IMergeTreeGroupMsg`
 - `IMergeTreeInsertMsg`
 - `IMergeTreeOp`
 - `IMergeTreeRemoveMsg`
 - `IRelativePosition`
 - `MergeTreeDeltaType`
 - `ReferenceType`
 - `MapLike`
 - `PropertySet`
 - `compareReferencePositions`
 - `DetachedReferencePosition`
 - `maxReferencePosition`
 - `minReferencePosition`
 - `ReferencePosition`
 - `refGetTileLabels`
 - `refHasTileLabel`
 - `refHasTileLabels`
 - `refTypeIncludesFlag`
 - `SegmentGroupCollection`
 - `PropertiesManager`
 - `PropertiesRollback`
 - `TextSegment`

</details>

For reviewers: is there anything else in the above list it makes sense to remove or mark `@internal`?

Things I think it would be nice to make `@internal`:
 - `LocalReferenceCollection` (all fields and methods are already `@internal`)
 - `PropertiesManager`
 - `PropertiesRollback`
 - `refGetTileLabels` and friends
 - `IMergeTreeInsertMsg` and friends
 - all other interfaces starting with `IMergeTree`
 - maybe `Client`?

Exports used by loop:

 - `reservedMarkerIdKey`
 - `ReferenceType`
 - `PropertySet`
 - `appendToMergeTreeDeltaRevertibles`
 - `revertMergeTreeDeltaRevertibles`
 - `discardMergeTreeDeltaRevertible`
 - `ISegment`
 - `MergeTreeDeltaRevertible`
 - `Marker`
 - `TrackingGroup`
 - `LocalReferencePosition`
 - `TextSegment`
 - `MapLike`
 - `MergeTreeDeltaType`

The above list is the exports found in the office-bohemia codebase. They are what I think should be re-exported from merge-tree by sequence.

### CODE REVIEW

File: packages/dds/sequence/src/intervals/sequenceInterval.ts [hunk 1]
```diff
@@ -341,7 +340,6 @@ export class SequenceInterval implements ISerializableInterval {
 		seq?: number,
 		op?: ICombiningOp,
 	): PropertySet | undefined {
-		this.initializeProperties();
```
Comment (anthony-murphy) [inline packages/dds/sequence/src/intervals/sequenceInterval.ts:344]: this change isn't like the others. i think it is probably good, but we should pull it out. i also wonder if property manager can be readonly.

File: api-report/merge-tree.api.md [hunk 2]
```diff
@@ -991,13 +936,13 @@ export class SegmentGroupCollection {
     constructor(segment: ISegment);
     // (undocumented)
     copyTo(segment: ISegment): void;
-    // (undocumented)
+    // @internal (undocumented)
```
Comment (anthony-murphy) [inline api-report/merge-tree.api.md:939]: no need to mark members internal on an internal type

File: packages/dds/merge-tree/src/index.ts [hunk 3]
```diff
@@ -133,7 +112,7 @@ export { SegmentGroupCollection } from "./segmentGroupCollection";
 export { PropertiesManager, PropertiesRollback } from "./segmentPropertiesManager";
 export { SortedSet } from "./sortedSet";
 export { SortedSegmentSet, SortedSegmentSetItem } from "./sortedSegmentSet";
-export { IJSONTextSegment, IMergeTreeTextHelper, TextSegment } from "./textSegment";
```
Comment (anthony-murphy) [inline packages/dds/merge-tree/src/index.ts:136]: the typing is not used, but we kind do export this type, see TextSegment.toJSONObject, i think i'd rather see that method return type explicitly typed, as we depend on round trip

File: api-report/merge-tree.api.md [hunk 4]
```diff
@@ -956,28 +907,22 @@ export function refTypeIncludesFlag(refPosOrType: ReferencePosition | ReferenceT
 // @public (undocumented)
 export const reservedMarkerIdKey = "markerId";
 
-// @public (undocumented)
+// @internal (undocumented)
```
Comment (anthony-murphy) [inline api-report/merge-tree.api.md:910]: i think the reserved keys are used by consumers, as they set those properties in segment annotations

Comment (anthony-murphy) [reply api-report/merge-tree.api.md:910]: you can see that in examples

File: api-report/merge-tree.api.md (hunk 2 above)
Comment (connorskees) [reply api-report/merge-tree.api.md:939]: I think the GitHub diff makes this look like it is part of `SegmentGroup`. This is a field on `SegmentGroupCollection`, which is public

File: api-report/merge-tree.api.md (hunk 4 above)
Comment (connorskees) [reply api-report/merge-tree.api.md:910]: I only found references to `reservedMarkerIdKey` in the fluid-bohemia repo when I looked, but it makes sense to be more conservative with these constants -- I was debating whether to remove them for some time. I've removed the `@internal` marker from all of them, though I think only 2 have usage in examples.

**Review by anthony-murphy — APPROVED** [review]

**Comment by connorskees** [comment]
Deprecation PR: https://github.com/microsoft/FluidFramework/pull/17952

I will resolve merge conflicts in this PR once the deprecation PR merges, as the deprecations will also conflict with this PR.

**Pull request merged**